crates/autopilot/src/domain/eth/mod.rs

```diff
@@ -1,4 +1,34 @@
-pub use primitive_types::{H160, U256};
+pub use primitive_types::{H160, H256, U256};
+
+/// An address. Can be an EOA or a smart contract address.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Address(pub H160);
+
+impl From<H160> for Address {
```

Comment:
We can derive this (and the one in L17) with `derive_more`

Reply:
will leave this to be implemented with https://github.com/cowprotocol/services/pull/2709

Reply:
@Mateo-mro will fix this in a separate PR as there are other occurrences in this very same file that should be all fixed together.

Reply:
https://github.com/cowprotocol/services/pull/2727

this PR:
abb505ab09e70e606da21dbf4500a94fdfcb335e